x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/function/scalar/math/ToNumberProcessorTests.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.expression.function.scalar.math;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.QlIllegalArgumentException;
+
+import static org.elasticsearch.xpack.ql.expression.function.scalar.FunctionTestUtils.l;
+import static org.elasticsearch.xpack.ql.tree.Source.EMPTY;
+
+
+public class ToNumberProcessorTests extends ESTestCase {
+
+    private static Object process(Object value, Object base) {
+        return new ToNumber(EMPTY, l(value), l(base)).makePipe().asProcessor().process(null);
+    }
+
+    private static String error(Object value, Object base) {
+        QlIllegalArgumentException saie = expectThrows(QlIllegalArgumentException.class,
+            () -> new ToNumber(EMPTY, l(value), l(base)).makePipe().asProcessor().process(null));
+        return saie.getMessage();
+    }
+
+    public void toNumberWithPositiveInteger() {
+        int number = randomIntBetween(0, 1000);
+
+        assertEquals(number, process(Integer.toString(number), null));
+        assertEquals(number, process("0x" + Integer.toHexString(number), null));
+
+        assertEquals(number, process(Integer.toString(number), 10));
+        assertEquals(number, process(Integer.toOctalString(number), 8));
+        assertEquals(number, process(Integer.toHexString(number), 16));
+        assertEquals(number, process("0x" + Integer.toHexString(number), 16));
+    }
+
+    public void toNumberWithNegativeInteger() {
+        int posInt = randomIntBetween(1, 1000);
+        int negInt = -posInt;
+
+        assertEquals(negInt, process(Integer.toString(negInt), null));
+
+        assertEquals(negInt, process(Integer.toString(negInt), 10));
+        assertEquals(negInt, process("-" + Integer.toOctalString(posInt), 8));
+        assertEquals(negInt, process("-" + Integer.toHexString(posInt), 16));
+
+        assertEquals(negInt, process("-0x" + Integer.toHexString(posInt), 16));
+    }
+
+    public void toNumberWithPositiveFloat() {
+        double number = randomDoubleBetween(0.0, 1000.0, true);
+
+        assertEquals(number, process(Double.toString(number), null));
+        assertEquals(number, process(Double.toString(number), 10));
+    }
+
+    public void toNumberWithNegativeFloat() {
+        double number = randomDoubleBetween(-1000.0, -0.1, true);
+
+        assertEquals(number, process(Double.toString(number), null));
+        assertEquals(number, process(Double.toString(number), 10));
+    }
+
+    public void toNumberWithMissingInput() {
+        assertNull(process(null, null));
+        assertNull(process(null, 8));
+        assertNull(process(null, 10));
+        assertNull(process(null, 16));
```

Comment:
`assertNull(process(123, null))` is missing.

Reply:
ah. that's because if the base is null, then the base will be autodetected.
I don't know if there's a way to mention that the parameter was not supplied vs it was evaluated as false. I believe substring() and a few other functions have similar behavior